tools/devtool

```diff
@@ -685,10 +685,27 @@ cmd_test() {
     say "RPM microcode_ctl version: $(rpm -q microcode_ctl)"
 
     env |grep -P "^(AWS_EMF_|BUILDKITE|CODECOV_)" > env.list
-    if [[ $performance_tweaks -eq 1 ]] && [[ "$(uname --machine)" == "x86_64" ]]; then
-      say "Detected CI and performance tests, tuning CPU frequency scaling and idle states for reduced variability"
+    if [[ $performance_tweaks -eq 1 ]]; then
+      if [[ "$(uname --machine)" == "x86_64" ]]; then
+        say "Detected CI and performance tests, tuning CPU frequency scaling and idle states for reduced variability"
+
+        apply_performance_tweaks
+      fi
+
+      # It seems that even if the tests using huge pages run sequentially on ag=1 agents, right-sizing the huge pages
+      # pool to the total number of huge pages used across all tests results in spurious failures with pool depletion
+      # anyway (something else on the host seems to be stealing our huge pages, and we cannot "ear mark" them for
+      # Firecracker processes). Thus, just allocate 4GB of them and call it a day.
+      say "Setting up huge pages pool"
+      num_hugetlbfs_pages=2048
+
+      # Allocate enough 2M hugetlbfs pages to boot a single 128MB microvm
+      huge_pages_old=$(cat /sys/kernel/mm/hugepages/hugepages-2048kB/nr_hugepages)
+      huge_pages_new=$(echo $num_hugetlbfs_pages |sudo tee /sys/kernel/mm/hugepages/hugepages-2048kB/nr_hugepages)
```

Comment:
Mh, all the tests in test_hugepages.py rely on this. I prefer doing it as ahead of time as possible (and doing it only one instead of allocated and de-allocating per test) to reduce the chance of the scenario where memory is too fragmented to allocated the huge pages we need (although, probably won't happen given how large ag=1 metals are).